interview-questions.md

```diff
@@ -8,48 +8,119 @@ Then, research the question to expand on your answer. Even if you feel you have
+JavaScript flavored markdown that behaves very similarly to HTML. JSX uses the syntax 'className' rather than HTML's 'class' to modify the
+look and behavior of a JSX element, this happens because class is already a protected word in JavaScript. JSX is a building block of React
+components and what allows React to create user interfaces.
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package that installs a React app on the folder you want to work in.
 
-Researched answer:
+Researched answer: Yarn is a package manager that manage node modules and create a React App.  A React app needs node modules (code snippet depedencies) organized correctly.
 
 3. What is a React component?
 
-Your answer:
+Your answer: Components are reusable and independant bits of code.
 
-Researched answer:
+Researched answer: In React components are independent and reusable bits of code. They serve the same purpose as a JavaScript functions,
+but work in isolation and return HTML. Components have 2 types: Class components and Function components.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: The difference between them is that 'state' is local and 'props' isn't just local.
 
-Researched answer:
+Researched answer: The difference between 'state' and 'props' is that state is internal and controlled by the component itselt while props are external and controlled by whatever renders the component.
 
 5. What is the DOM?
 
-Your answer:
+Your answer: Document object Model is a programming interface.
 
-Researched answer:
+Researched answer: The DOM or Document Object Model is a programming interface for web documents, that represents the page so that programs can change the document structure, style and content. The DOM represents the document as nodes and objects, that way, programming languages can interact with the page. The DOM is not part of the JavaScript languages, but is instead a Web API used to build websites, it was designed to be independent of any particular programming language, making the structural representation of the document available from a single, consistent API. 
 
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: I do not know.
 
-Researched answer:
+Researched answer: In HTML, 'div' and 'span' tags are elements used to define parts of a document, so they are identifiable when a unique classification is necessary. Where other HTML elements such as 'p(paragraph)' 'em(emphasis)' accurately represent the semantics of the content, the additional use of 'span' and 'div' tagas leads to better accessibility for readers and easier maintanability for authors. Where no existing HTML element is applicable, 'span' and 'div' can valuably represent parts of a document so that HTML atributes as 'class','id','lang' or 'div' can be applied. 
```

Comment:
Your researched answers are for you to research the different topics and concepts, not to take things word for word on the internet and repeat it here. When you just copy and paste, you are not helping yourself learn as much as you can.